tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderSteps.cs

```diff
@@ -1,187 +1,202 @@
+            {
+                if (ordersTableItem.OrganisationAddressPostcode != null)
+                {
+                    ordersTableItem.OrganisationAddressId = GetIdFromContext(ScenarioContextKeys.AddressMapDictionary,
+                        ordersTableItem.OrganisationAddressPostcode);
+                }
+
+                if (ordersTableItem.OrganisationContactEmail != null)
+                {
+                    ordersTableItem.OrganisationContactId = GetIdFromContext(ScenarioContextKeys.ContactMapDictionary,
+                        ordersTableItem.OrganisationContactEmail);
+                }
+
+                var order = OrderEntityBuilder
+                    .Create()
+                    .WithOrderId(ordersTableItem.OrderId)
+                    .WithDescription(ordersTableItem.Description)
+                    .WithOrganisationId(ordersTableItem.OrganisationId)
+                    .WithOrganisationName(ordersTableItem.OrganisationName)
+                    .WithOrganisationOdsCode(ordersTableItem.OrganisationOdsCode)
+                    .WithOrganisationAddressId(ordersTableItem.OrganisationAddressId)
+                    .WithOrganisationBillingAddressId(ordersTableItem.OrganisationBillingAddressId)
+                    .WithOrganisationContactId(ordersTableItem.OrganisationContactId)
+                    .WithOrderStatusId(ordersTableItem.OrderStatusId)
+                    .WithDateCreated(ordersTableItem.Created)
+                    .WithLastUpdatedBy(ordersTableItem.LastUpdatedBy)
+                    .WithLastUpdatedName(ordersTableItem.LastUpdatedByName)
+                    .WithLastUpdated(ordersTableItem.LastUpdated)
+                    .Build();
+
+                await order.InsertAsync(_settings.ConnectionString);
+            }
+        }
+
+        [When(@"a GET request is made for a list of orders with organisationId (.*)")]
+        public async Task WhenAGETRequestIsMadeForOrders(Guid organisationId)
+        {
+            await _request.GetAsync(string.Format(_orderOrganisationsUrl, organisationId));
+        }
+
+        [Then(@"the orders list is returned with the following values")]
+        public async Task ThenTheOrdersListIsReturnedWithTheFollowingValues(Table table)
+        {
+            var expectedOrders = table.CreateSet<GetOrdersTable>();
+
+            var orders = (await _response.ReadBodyAsJsonAsync()).Select(CreateOrders);
+
+            orders.Should().BeEquivalentTo(expectedOrders);
+        }
+
+        [Then(@"an empty list is returned")]
+        public async Task AnEmptyListIsReturned()
+        {
+            var orders = (await _response.ReadBodyAsJsonAsync()).Select(CreateOrders);
+            orders.Count().Should().Be(0);
+        }
+
+        [Then(@"the order with orderId (.*) is updated in the database with data")]
+        [Then(@"the order is created in the database with orderId (.*) and data")]
```

Comment:
Why have you added an extra statement, can you use one or the other, not have both

Reply:
The step is used in a create feature file and a update feature file with the same implementation. I added the extra tag so that it read better in the file it was being used in.  I can duplicate the implementation again so the steps are independent.